doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,235 @@
+"""
+============================================
+Restore spotted cornea image with inpainting
+============================================
+
+Optical coherence tomography (OCT) is a non-invasive imaging technique used by
+ophthalmologists to take pictures of the back of a patient's eye [1]_.
+When performing OCT,
+dust may stick to the reference mirror of the equipment, causing dark spots to
+appear on the images. The problem is that these dark spots cover areas of
+in-vivo tissue, hence hiding data of interest. Our goal here is to restore
+(reconstruct) the hidden areas based on the pixels near their boundaries.
+
+This tutorial is adapted from an application shared by Jules Scholler [2]_.
+The images were acquired by Viacheslav Mazlin.
+
+.. [1] David Turbert, reviewed by Ninel Z Gregori, MD (2023)
+       `What Is Optical Coherence Tomography?
+       <https://www.aao.org/eye-health/treatments/what-is-optical-coherence-tomography>`_,
+       American Academy of Ophthalmology.
+.. [2] Jules Scholler (2019) "Image denoising using inpainting"
+       https://www.jscholler.com/2019-02-28-remove-dots/
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import plotly.io
+import plotly.express as px
+import plotly.graph_objects as go
+from scipy import sparse
+
+from skimage import (
+    filters, measure, morphology, restoration
+)
+from skimage.data import palisades_of_vogt
```

Comment:
Oh, what's the reference for this new recommendation, please?

Reply:
What kind of reference are you looking for? As far as I'm aware the old approach was a limitation of our old import mechanism and namespace hierarchy. With the lazy loading added we no longer have do that and there are big advantages to the new way that make it worth it in my opinion. 
- It makes the origin of a function, class, etc. very clear at the place where it is actually used.
- It doesn't pollute the current namespace with often generic and ambiguous terms as "data", "filters", "io", ... which are also prone to naming conflicts with similar libraries. E.g. `label`, is it from scikit-image or SciPy? 
- For me it's also a nicer experience when developing or reading code, no need to jump between imports and code as I am adding new functionality from scikit-image.

Reply:
> From past discussions I got the feeling that everyone agrees that the new approach is the one to recommend.

Fine, then it would mean that `import skimage as ski` has become a *de facto* standard! It's just that I missed that episode, so I was wondering if there was any

> What kind of reference are you looking for? 

blog post, discussion on GitHub, discussion on the forum, etc. where this was 'established' in some way.

> It doesn't pollute the current namespace with often generic and ambiguous terms as "data", "filters", "io", ... which are also prone to naming conflicts with similar libraries. E.g. label, is it from scikit-image or SciPy?

I really care about the module level for educational reasons (cf. https://github.com/scikit-image/scikit-image/pull/6853/files#r1182903727), but you're saying that even module names may be shared throughout libraries? I hear you.

Reply:
Yes! And as far as I understand your https://github.com/scikit-image/scikit-image/pull/5365#issuecomment-855788844, the `ski.filters.gaussian` approach totally works with your argument in favor of importing submodules too :blush: :

> I find it's more pedagogic to import submodules rather than functions directly, so that when the reader runs or reads the code, they see feature.corner_harris, filters.gaussian, etc. and they think "Gaussian is a filter indeed" and, indirectly, they learn the codebase organization and, also, the import header is then more compact.

Reply:
We should really start updating our documentation in this regard!